enterprise/server/cmd/ci_runner/main.go

```diff
@@ -1421,22 +1420,31 @@ func (ws *workspace) sync(ctx context.Context) error {
 	for _, path := range *gitCleanExclude {
 		cleanArgs = append(cleanArgs, "-e", path)
 	}
-	if err := git(ctx, ws.log, cleanArgs...); err != nil {
+	if _, err := git(ctx, ws.log, cleanArgs...); err != nil {
 		return err
 	}
 
 	// Create the local branch if it doesn't already exist, then update it to point to the checkout ref
-	if err := git(ctx, ws.log, "checkout", "--force", "-B", checkoutLocalBranchName, checkoutRef); err != nil {
+	if _, err := git(ctx, ws.log, "checkout", "--force", "-B", checkoutLocalBranchName, checkoutRef); err != nil {
 		return err
 	}
 
+	// If commit sha is not set, pull the sha that is checked out so that it can be used in Github Status reporting
+	if *commitSHA == "" {
+		headCommitSha, err := git(ctx, ws.log, "rev-parse", "HEAD")
+		if err != nil {
+			return err
+		}
+		commitSHA = &headCommitSha
```

Comment:
uber nit: this deletes the flag pointer, meaning that a subsequent `flag.Parse()` wouldn't populate `commitSHA` as expected. It might be safer to write `*commitSHA = headCommitSha`. I don't feel strongly about it though.